test/e2e/framework/metrics/metrics_grabber.go

```diff
@@ -235,16 +238,26 @@ func (g *Grabber) Grab() (Collection, error) {
 	return result, nil
 }
 
-func (g *Grabber) getMetricsFromPod(client clientset.Interface, podName string, namespace string, port int) (string, error) {
-	rawOutput, err := client.CoreV1().RESTClient().Get().
-		Namespace(namespace).
-		Resource("pods").
-		SubResource("proxy").
-		Name(fmt.Sprintf("%v:%v", podName, port)).
-		Suffix("metrics").
-		Do(context.TODO()).Raw()
-	if err != nil {
+func (g *Grabber) getMetricsFromPod(client clientset.Interface, podName string, namespace string, port int) (metrics string, err error) {
+	pollErr := wait.PollImmediate(5*time.Second, 2*time.Minute, func() (bool, error) {
+		rawOutput, err := client.CoreV1().RESTClient().Get().
+			Namespace(namespace).
+			Resource("pods").
+			SubResource("proxy").
+			Name(fmt.Sprintf("%v:%v", podName, port)).
+			Suffix("metrics").
+			Do(context.TODO()).Raw()
+		if err != nil {
+			if testutils.IsRetryableAPIError(err) {
```

Comment:
is this needed and catch current problem or we just should retry on error?

Reply:
SGTM

Reply:
although a question here is when this happens:
> the getMetricsFromPod function tries to Get metrics from one pod,
however this pod may no be ready at that moment and it fails the
test.

does `IsRetryableAPIError(err) == true`?

Reply:
is a 404, we need to retry always